mlflow/exceptions.py

```diff
@@ -67,6 +67,8 @@ class MlflowException(Exception):
     The error message associated with this exception may be exposed to clients in HTTP responses
     for debugging purposes. If the error text is sensitive, raise a generic `Exception` object
     instead.
+
+    Note: Instance of `MlflowException` is pickleable, see `__getstate__` and `__setstate__`
```

Comment:
`MlflowException` is already picklable:

```
>>> import mlflow
>>> import pickle
>>> pickle.dumps(mlflow.MlflowException("hello"))
```

Reply:
No. when `pickle.load` it fails.

Reply:
You can repro it by the following code:

```
import mlflow.exceptions
import pickle
v1 = mlflow.exceptions.RestException({"error_code": "INTERNAL_ERROR", "message": "abc"})
s1 = pickle.dumps(v1)
v2 = pickle.loads(s1)  # error occurs
```

Error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/weichen.xu/work/mlflow/mlflow/exceptions.py", line 117, in __init__
    error_code = json.get("error_code", ErrorCode.Name(INTERNAL_ERROR))
AttributeError: 'str' object has no attribute 'get'
```